### DISCUSSION TITLE
G7 connection loss

### DISCUSSION THREAD

**Discussion by jgslade** [NightscoutFoundation/xDrip#3669] [Bug reports] [answered]
I have been having connection issues with the Dexcom G7, it'll start some time during the first 3 days of a sensor, but once it it'll last the life of the sensor. The sensor will consistently connect to the receiver and if I switch to the Dexcom app it will consistently connect to the sensor, I don't pair the sensor to the Dexcom app until after I start getting connection issues with Xdrip. I have not had any connection issues to my Omnipod Dash with AAPS, or to my smart watch or headphones, so I don't think there is a bluetooth problem with my phone.

I started a new sensor yesterday and got my first connection issue with it. I have enabled ob1 debugged and submitted the logs. The connection failures occurred at 15:42, 15:47, and 15:52, at 15:57 I was asked to pair the sensor again.

My setup:

Pixel 8 Pro
Dexcom G7
Omnipod Dash

**Comment by Navid200** [comment, marked as answer]
Some Pixel users are not aware how important it is to disable Bluetooth battery optimization in addition to xDrip battery optimization.
Have you done that?

![Screenshot_20240917-182520](https://github.com/user-attachments/assets/bfb49b00-ccfb-4f44-b458-4e30fda80e2d)
![Screenshot_20240917-182730](https://github.com/user-attachments/assets/809b7523-bba0-4079-a7c7-5ae70746c5b4)

**Reply by jgslade** [reply]
Is this a recent change, I got my pixel last October, and these connection issues started this summer?

Xdrip was set to unrestricted but legacy bluetooth was not.

**Reply by Navid200** [reply]
No, this is not a recent change.

**Reply by Navid200** [reply]
But, G7 connectivity (compared to G6) has some issues and we are working on them.

**Reply by jgslade** [reply]
Guess I was lucky for the first 6 months or so with the pixel, lol.

**Reply by Navid200** [reply]
I switched to Pixel about a year ago and I couldn't be happier compared to Motorola I was using before.  Relatively speaking, it is a solid performer.  I was not suggesting that Pixel was bad.

**Reply by jgslade** [reply]
I meant with the incorrect setting, I'm loving my pixel, and other than this weird connection issue had no complaints. It a fantastic phone for DIY looping.

I'm also really liking the G&, been on it for a bit over a year, and the no downtime in reading is fantastic, when it connects... But the companion mode in xdrip is a great life saver for when I did get connection issues.

**Reply by Navid200** [reply]
I have a Pixel 6a.  
I also use G7.
It does not disconnect.
HOWEVER, there are times I need to test something in order to recreate a problem reported by someone.  For example today someone claimed that companion app was not working.  So, I stopped collection by xDrip and switched to Dexcom app.  OK, then it was not easy to reconnect when I was done with testing.  It took me a while to get it back on track.  

Oh, and I am running Android 15 beta to make sure that we will not have any surprises when 15 is released in a few weeks.  Although it has been supposed to be released in a few weeks for a few months now!

**Reply by jgslade** [reply]
I tried once to switch back to see if xdrip would collect again, but it just kept erroring, so I was only try when starting a new sensor now, but no connection issues since changing that setting, so here's hoping that was my problem.

**Reply by Lloansi** [reply]
Hi @Navid200 , I'm using a Samsung s23 FE and like @jgslade I have been debugging the app since it connects and disconnects all the time, without me being able to see the data in real time in the xDrip app, in the log I have seen that it correctly receives the sensor information (name, mac ...) and when I get that it is connected, then it becomes disconnected immediately.

I understand that the problem is in starting the sensor from the dexcom app (which is how I did it), could you tell me step by step how you have done to reconnect correctly and read the sensor data in real time in the xDrip app ?

I also understand that to connect it to xDrip, I must uninstall the dexcom app, and once it is connected correctly do not reinstall the dexcom g7 app because if not, the same thing will happen again, right?

**Reply by Navid200** [reply]
I insert a new sensor usually when I still have several hours left on my current sensor.
After inserting, I use a different device other than my main to connect to it using xDrip to confirm that it is not a faulty sensor and is providing readings.  After I see that, I stop using the extra device.

When my current sensor stops working or before depending on my needs, I peel off my old sensor and put it inside a metallic box.  I change the transmitter ID in xDrip on my main phone to that of the new device.  My main phone then establishes connectivity for the first time to my new device and shows readings.  

For someone who reads posts asking for help, may be read 10s of different posts by 10s of different people in different forums, it is very important to be able to remind himself/herself what each thread is exactly about.
That's why I always ask people asking for help to include the crucuial info in the very first post.  So, if I see a new post in a thread I am attending to and I have forgotten the details, I can easily look at the very first post.
This methodology falls apart if  people post in existing threads asking for help.  
So, I always ask people to please do not post in someone else's thread asking for help.
Mind you, there are times we open a thread and ask for feedback about a particular problem.  That is different.

Maintaining connectivity to G7 from xDrip can be interrupted on different phones for different reasons, even on the same phones.  

So, if you now respond to this post and go into detail about your problem on your Samsung phone, this thread will become to convoluted!  And it will only get worse the more each one of you posts in it.

**Reply by jgslade** [reply]
It's been around 24 hours since I made the change and not had any connection issues. Thanks so much for the fix Navid.
